rust/pact_models/src/path_exp.rs

```diff
@@ -17,6 +18,59 @@ pub enum PathToken {
   StarIndex
 }
 
+fn matches_token(path_fragment: &str, path_token: &PathToken) -> usize {
+  match path_token {
```

Comment:
I am assuming this is all just the code moved from the matchingrules.

Reply:
Yes indeed; it may be clearer to see if you look only at the first commit's diff.